ui/src/test/java/edu/wpi/grip/ui/codegeneration/tools/PipelineInterfacer.java

```diff
@@ -0,0 +1,65 @@
+package edu.wpi.grip.ui.codegeneration.tools;
+import static org.junit.Assert.fail;
+
+import java.lang.reflect.InvocationTargetException;
+
+public class PipelineInterfacer {
+	private Class pipeline;
+	private Object instance;
+	public PipelineInterfacer(String className){
+		pipeline = PipelineCreator.makeClass(className);
+		try {
+			instance = pipeline.getConstructor().newInstance();
+		} catch (InstantiationException | IllegalAccessException
+				| IllegalArgumentException | InvocationTargetException
+				| NoSuchMethodException | SecurityException e) {
+			e.printStackTrace();
+			fail("Failure to instansiate class");
+		}
+	}
+	public void setSource(int num, Object value){
+		try{
+			pipeline.getMethod("setsource"+num, value.getClass()).invoke(instance, value);
+		}catch (NoSuchMethodException | SecurityException | IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
+			e.printStackTrace();
+			fail("setsource"+num +" is not valid for class "+value.getClass().getSimpleName());
+		}
+	}
+	public void process(){
+		try {
+			pipeline.getMethod("process").invoke(instance);
+		} catch (IllegalAccessException | IllegalArgumentException
+				| InvocationTargetException | NoSuchMethodException
+				| SecurityException e) {
+			e.printStackTrace();
+			fail("process method doesn't exist");
+		}
+	}
+	public Object getOutput(int num){
+		try{
+			return pipeline.getMethod("getoutput"+num).invoke(instance);
```

Comment:
Yes but no. Yes that is the way it should be. No because the actual method that our generator is generating is getoutput.